src/Psl/DateTime/Timestamp.php

```diff
@@ -0,0 +1,305 @@
+
+        // No manual normalization required here due to fromRaw handling it
+        return self::fromRaw($newSeconds, $newNanoseconds);
+    }
+
+    /**
+     * Subtracts the specified duration from this timestamp object, returning a new instance with the subtracted duration.
+     *
+     * @throws Exception\UnderflowException If subtracting the duration results in an arithmetic underflow.
+     * @throws Exception\OverflowException If subtracting the duration results in an arithmetic overflow.
+     *
+     * @mutation-free
+     */
+    public function minus(Duration $duration): static
+    {
+        [$h, $m, $s, $ns] = $duration->getParts();
+        $totalSeconds = SECONDS_PER_MINUTE * $m + SECONDS_PER_HOUR * $h + $s;
+        $newSeconds = $this->seconds - $totalSeconds;
+        $newNanoseconds = $this->nanoseconds - $ns;
+
+        // No manual normalization required here due to fromRaw handling it
+        return self::fromRaw($newSeconds, $newNanoseconds);
+    }
+
+    /**
+     * Formats this {@see TemporalInterface} instance based on specified styles, pattern, timezone, and locale.
+     *
+     * This method provides a flexible way to format the date and time representation of this {@see TemporalInterface} object,
+     * allowing for customization through various parameters. If parameters are not provided, default values are used,
+     * ensuring a sensible output even when minimal information is specified.
+     *
+     * Usage example:
+     *
+     * ```php
+     * use Psl\DateTime;
+     * use Psl\Locale;
+     * use Psl\IO;
+     *
+     * $timestamp = DateTime\Timestamp::now();
+     * $formatted = $timestamp->format(locale: Locale\Locale::Arabic);
+     *
+     * IO\write_line($formatted); //
+     * ```
+     *
+     * @param null|FormatDateStyle $date_style Specifies the style of the date portion for parsing. Defaults to {@see DateFormatStyle::default()}.
+     * @param null|FormatTimeStyle $time_style Specifies the style of the time portion for parsing. Defaults to {@see TimeFormatStyle::default()}.
+     * @param null|FormatPattern|string $pattern Specifies a custom format pattern to be used for parsing. A default, implementation-specific pattern is used if null.
+     * @param null|Timezone $timezone Determines the timezone for which the date and time are formatted. Defaults to {@see Timezone::default()}.
+     * @param null|Locale $locale Specifies the locale to be used for formatting locale-specific date and time components. Defaults to {@see Locale::default()}.
+     *
+     * @return string The formatted date and time string.
+     *
+     * @see https://unicode-org.github.io/icu/userguide/format_parse/datetime/#datetime-format-syntax
+     *
+     * @mutation-free
+     */
+    public function format(null|FormatDateStyle $date_style = null, null|FormatTimeStyle $time_style = null, null|FormatPattern|string $pattern = null, null|Timezone $timezone = null, null|Locale $locale = null): string
```

Comment:
this is probably something we need to document: there are a lot of options.
Does it make sense to have both day / time styles and patterns in the same function?

Reply:
I thought about having:

```
format($pattern, $timezone, $locale)
toString($date_style, $time_style, $timezone, $locale)
__toString() ( alias for toString(null, null, null, null) ).
```

but not sure. 🤔

Reply:
Something like that could be nice indeed. Don't think we really need `__toString()`, cause that wont take any params.

Reply:
I think `__toString()` is okay to keep, as it allows for `echo $temporal;`